src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php

```diff
@@ -144,8 +145,16 @@ protected function execute(InputInterface $input, OutputInterface $output)
             ? new TargetOperation($currentCatalogue, $extractedCatalogue)
             : new MergeOperation($currentCatalogue, $extractedCatalogue);
 
+        $domains = $operation->getDomains();
+        $domain = $input->getOption('domain');
+
+        if (null !== $domain) {
+            $domains = array_intersect($domains, array($domain));
```

Comment:
I don't think we need this line: when the option is set, let's call setDomains with it